test/e2e/framework/cluster.go

```diff
@@ -0,0 +1,330 @@
+// Copyright (C) 2024 ScyllaDB
```

Comment:
Where is the `ClusterInterface` and the assertion that both the `Cluster` and `Framework` implement it?

Reply:
must have missed it, although I could swear I was adding it, maybe lost it during rebasing somehow... anyway added